azure/packages/azure-client/src/test/AzureClient.spec.ts

```diff
@@ -164,4 +178,95 @@ describe("AzureClient", () => {
+     * to start the connection in write mode, but if the write permission is not available or the file is read
+     * only, mode will be read.
+     *
+     * Expected behavior: The container should start with the connectionMode in `write`.
+     */
+    it("can create a container in write mode", async () => {
+        const { container } = await client.createContainer(schema);
+        const containerId = await container.attach();
+        await timeoutPromise((resolve) => container.once("connected", resolve), {
+            durationMs: 1000,
+            errorMsg: "container connect() timeout",
+        });
+
+        const { container: containerGet } = await client.getContainer(containerId, schema);
+
+        assert.strictEqual(
+            connectionModeOf(container),
+            "write",
+            "Creating a container is not in write mode",
+        );
+        assert.strictEqual(
+            connectionModeOf(containerGet),
+            "write",
+            "Getting a container is not in write mode",
+        );
+    });
+
+    /**
+     * Scenario: Test if AzureClient with only read permission starts the container in read mode.
+     * AzureClient will attempt to start the connection in write mode, but if the write permission
+     * is not available or the file is read only, mode will be read.
+     *
+     * Expected behavior: The container should start with the connectionMode in `read`.
+     */
+    it("can create a container with only read permission in read mode", async () => {
+        const readOnlyAzureClient = createAzureClientWithScopes([ScopeType.DocRead]);
+
+        const { container } = await readOnlyAzureClient.createContainer(schema);
+        const containerId = await container.attach();
+        await timeoutPromise((resolve) => container.once("connected", resolve), {
+            durationMs: 1000,
+            errorMsg: "container connect() timeout",
+        });
+        const { container: containerGet } = await readOnlyAzureClient.getContainer(containerId, schema);
+
+        assert.strictEqual(
+            connectionModeOf(container),
+            "read",
+            "Creating a container with only read permission is not in read mode",
+        );
+
+        assert.strictEqual(
+            connectionModeOf(containerGet),
+            "read",
+            "Getting a container with only read permission is not in read mode",
+        );
+    });
+
+    /**
+     * Scenario: Test if AzureClient with only write permission starts the container in write mode.
+     * AzureClient will attempt to start the connection in write mode, but if the write permission
+     * is not available or the file is read only, mode will be read.
+     *
+     * Expected behavior: The container should start with the connectionMode in `write`.
+     */
+    it("can create a container with only write permission in write mode", async () => {
+        const writeOnlyAzureClient = createAzureClientWithScopes([ScopeType.DocWrite]);
```

Comment:
For your cases it may not matter, but generally you may need both ScopeType.DocRead & ScopeType.DocWrite.

Reply:
added read permission